src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -60,7 +60,10 @@ public InternalUsersApiAction(final Settings settings, final Path configPath, fi
             ThreadPool threadPool, AuditLog auditLog) {
         super(settings, configPath, controller, client, adminDNs, cl, cs, principalExtractor, evaluator, threadPool,
                 auditLog);
+        registerHandlers(controller);
+    }
 
+    protected void registerHandlers(RestController controller) {
```

Comment:
let's create abstract method in base class and override it in all child classes as discussed.